src/fxos/getProcesses.js

```diff
@@ -0,0 +1,67 @@
+let fs = require('fs');
+
+const EOL = /\r\n?|\n/g;
+const BOUNDARY = /\b([A-Za-z0-9_\(\)]+)/g;
+
+function format(value) {
+  value = value.trim();
+  try {
+    return JSON.parse(value);
+  } catch (e) {
+    return value;
+  }
+}
+
+function formatName(name) {
+  return name
+    .replace('_', '')
+    .replace('(s)', 's')
+    .toLowerCase();
+}
+
+module.exports = async function getProcesses(adb, pid) {
+  let [content] = await adb.shell('b2g-info');
+  let lines = content.split(EOL).slice(1);
+  let rows = lines.slice(1, lines.indexOf(''));
+  let headings = lines[0];
+  let columns = [];
+  let match;
+
+  while (match = BOUNDARY.exec(headings)) {
+    columns.push({name: match[0], index: match.index});
+  }
+
+  let procs = rows.map(row => {
+    let proc = {};
+
+    columns.forEach((column, index) => {
+      // All columns but the last are right-aligned
+      let name = formatName(column.name);
+      let end = column.index + column.name.length;
+
+      // First column starts at the beginning of the string
```

Comment:
I'm very anti-`switch`, so I will change it at your request, otherwise I will leave it. :smile: